src/breedsRepo.js

```diff
@@ -1,82 +1,170 @@
-        const idMatchArray = breedsCopy.filter((breed) => {  // could use findIndexById instead of array.filter
-            return breed['breedId'] == id; 
-        })
-        return idMatchArray[0];
+
+        const dao2 = this.dao;
+        const checkForId = this.checkForId; 
+
+        const getProm = async function () {
+            try {
+                if (!checkForId(id)) {
+                    return null;
+                } else {
+                    const getBreed = await dao2.get('SELECT * FROM breeds WHERE breedId = ?;', [id]);
+                    return getBreed; 
+                };
+            } catch (err) {
+                console.log(err);
+            }
+        }
+        return getProm();
     }
 
     updateBreedById(id, object) {
-        const breedsCopy = this.getAllBreeds; 
-        const index = this.findIndexById(id); // 
-        if (index === -1) {
-            console.log(`index not present: ${index}`);
-            return null;
-        } else {
-            for (let key in object) {
-                if (breedsCopy[index][key] !== object[key]) {
-                    breedsCopy[index][key] = object[key];
-                    console.log(`Key '${key}' successfully updated`);
-                }
+        
+        const dao2 = this.dao;
+        const checkForId = this.checkForId; 
+        const getBreedById = this.getBreedById; 
+
+        const getProm = async function () {
+            try {                
+                if (!checkForId(id)) {
+                    return null;
+                } else {
+                    // iterates through object to create a string of keys and values in format: 'column = value column2 = value2'
+                    let arr = []; 
+                    console.log('object:' + JSON.stringify(object));
+                    for (let key in object) {
+                        arr.push(` ${key} = '${object[key]}'`);   
+                    }
+                    console.log('arr: ' + arr); 
+                    let sql = `UPDATE breeds SET ${arr.join()} WHERE breedId = ?;`
+                    await dao2.run(sql,[id]);
+                    let checkNewRecord = getBreedById(id);  
+                    return checkNewRecord; 
+                };
+            } catch (err) {
+                console.log(err); 
             }
-            this.setBreeds = breedsCopy; 
-            return breedsCopy[index]; 
-        };       
+        }
+        return getProm();      
     }
 
-    addBreed(newBreed) { 
-        const breedsCopy = this.getAllBreeds;
-        const newIndex = breedsCopy.length + 1
-        newBreed.breedId = newIndex;
-        breedsCopy.push(newBreed);
-        this.setBreeds = breedsCopy; 
-        return newBreed; 
+    addBreed(object) { 
+
+        const dao2 = this.dao;
+        const getBreedById = this.getBreedById; 
+
+        const getProm = async function () {
+            try {
+                // generates new id value. 
```

Comment:
SQLite does provide auto-incrementing IDs for a primary key. Primary keys are defined on a field which is known to be unique for all rows in the database and are used to help index rows which speeds up searches.

I'm not convinced that this step is necessary, as I inspected the database using `sqlite3 tables.db` and ran `.schema cats` which showed that the ID column is a primary key, so will auto increment. It's not too big a deal though and could be part of future work.
https://www.sqlite.org/faq.html#q1

Reply:
Thanks for letting me know about that! Although SQLite recommends that you should not use AUTOINCREMENT attribute because:

    'The AUTOINCREMENT keyword imposes extra CPU, memory, disk space, and disk I/O overhead and should be avoided if not strictly needed. It is usually not needed.'

Would my code block be more or less performance-heavy than using AUTOINCREMENT?

Reply:
I revoke my last comment! Having delved deeper, you are correct, the auto-incrementing of a PRIMARY KEY's rowid value is inherent and does what I need, which is searches for unused integers and uses them. The AUTOINCREMENT column attribute is another thing and we would NOT to looking to use that.

Reply:
I will incorporate this at a later date.